docs/docs/icicle/golang-bindings/msm.md

```diff
@@ -0,0 +1,140 @@
+# MSM
+
+
+### Supported curves
+
+`bls12-377`, `bls12-381`, `bn-254`, `bw6-761`
+
+## MSM Example
+
+```go
+package main
+
+import (
+    "github.com/stretchr/testify/assert"
+
+    "github.com/ingonyama-zk/icicle/wrappers/golang/core"
+    cr "github.com/ingonyama-zk/icicle/wrappers/golang/cuda_runtime"
+)
+
+func Main() {
+    // Obtain the default MSM configuration.
+    cfg := GetDefaultMSMConfig()
+    
+    // Define the size of the problem, here 2^18.
+    size := 1 << 18
+
+    // Generate scalars and points for the MSM operation.
+    scalars := GenerateScalars(size)
+    points := GenerateAffinePoints(size)
+
+    // Create a CUDA stream for asynchronous operations.
+    stream, _ := cr.CreateStream()
+    var p Projective
+    
+    // Allocate memory on the device for the result of the MSM operation.
+    var out core.DeviceSlice
+    _, e := out.MallocAsync(p.Size(), p.Size(), stream)
+    // Check for errors in memory allocation.
+    assert.Equal(t, e, cr.CudaSuccess, "Allocating bytes on device for Projective results failed")
+    
+    // Set the CUDA stream in the MSM configuration.
+    cfg.Ctx.Stream = &stream
+    cfg.IsAsync = true
+    
+    // Perform the MSM operation.
+    e = Msm(scalars, points, &cfg, out)
+    // Check for errors in the MSM operation.
+    assert.Equal(t, e, cr.CudaSuccess, "Msm failed")
+    
+    // Allocate host memory for the results and copy the results from the device.
+    outHost := make(core.HostSlice[Projective], 1)
+    outHost.CopyFromDevice(&out)
```

Comment:
```suggestion
    outHost := make(core.HostSlice[Projective], 1)
    cr.SynchronizeStream(&stream)
    outHost.CopyFromDevice(&out)
```
Synchrnizing the stream before copying back to device is necessary when `IsAsync = true`